### PR TITLE
fix: Update Partner Offer container mutation error logic

### PR DESCRIPTION
**Pull request by MrSltun**
This PR resolves [] 

### Description
This PR updates the conditions in the Partner Offer Container to go back in case the mutation errors out with something other than `not_acquireable` or `expired_partner_offer`

### Video
| Before | After |
|---|---|
| <video src="https://github.com/artsy/eigen/assets/20655703/c928ec2c-0631-43c7-b283-ce54c1c4dc77" /> | <video src="https://github.com/artsy/eigen/assets/20655703/c359255f-a421-4f1f-ac07-a4f7ed517263" /> |
	
### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Update partner offer container mutation error logic - mrsltun

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Update partner offer container mutation error logic - mrsltun)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8c9acab993dd16b8e0f48d4dccc98eaec75edaae
</p>

File: src/app/Scenes/PartnerOffer/PartnerOfferContainer.tsx [hunk 1]
```diff
@@ -29,6 +29,8 @@ export const PartnerOfferContainer: React.FC<{ partnerOfferID: string }> = ({ pa
         })
 
         return
+      } else {
+        goBack()
```
Comment (leamotta) [inline src/app/Scenes/PartnerOffer/PartnerOfferContainer.tsx:33]: giphy (3)

**Review by leamotta — DISMISSED** [review]

**Review by gkartalis — DISMISSED** [review]
Lets merge and 🍒  my only concern is that it would be way nicer to show a toast with an error or something and not redirect the user back 🤔 but I leave this up to emerald to decide

**Comment by damassi** [comment]
+1 to @gkartalis's comment. This change almost feels more like a bug than the bug itself, just in the sense that the user has tapped an item, the screen transitions, but then it kicks them back. Its likely they'll keep trying to click, and keep triggering error, and keep transitioning. For forever!  

Having a toast should be the minimum requirement before merge, or maybe having a reload button with a message (don't we have an error screen like that?)

**Review by iskounen — APPROVED** [review]

**Review by damassi — APPROVED** [review]

**Pull request merged**